js/apps/system/_admin/aardvark/APP/frontend/js/views/graphManagementView.js

```diff
@@ -513,46 +602,73 @@
     },
 
     setFromAndTo: function (e) {
+      // This method triggers as soon as an edgeDefinition name is either being added or removed.
       e.stopPropagation();
-      var map = this.calculateEdgeDefinitionMap();
       var id;
 
-      const smartGraphActive = $('#tab-smartGraph').parent().hasClass('active');
-      const enterpriseGraphActive = $('#tab-enterpriseGraph').parent().hasClass('active');
-
-      if (smartGraphActive || enterpriseGraphActive) {
-        if (e.added) {
-          if (this.eCollList.indexOf(e.added.id) === -1 && this.removedECollList.indexOf(e.added.id) !== -1) {
-            id = e.currentTarget.id.split('row_newEdgeDefinitions')[1];
-            $('input[id*="newEdgeDefinitions' + id + '"]').select2('val', null);
-            $('input[id*="newEdgeDefinitions' + id + '"]').attr(
-              'placeholder', 'The collection ' + e.added.id + ' is already used.'
-            );
-            return;
-          }
-          this.removedECollList.push(e.added.id);
-          this.eCollList.splice(this.eCollList.indexOf(e.added.id), 1);
-        } else {
-          if (e.removed) {
-            // TODO edges not properly removed within selection
-            this.eCollList.push(e.removed.id);
-            this.removedECollList.splice(this.removedECollList.indexOf(e.removed.id), 1);
+      const getAllCurrentEdgeDefinitionNames = () => {
+        let edgeDefNames = [];
+        for (let i = 0; i <= this.counter; i++) {
+          let name = _.pluck($('#s2id_newEdgeDefinitions' + i).select2('data'), 'text')[0];
```

Comment:
```suggestion
          let name = _.get($('#s2id_newEdgeDefinitions' + i).select2('data'), [0, 'text']);
```
Since we're only interested in the `text` property of the first element.